src/components/data-table/DataTableBody.tsx

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+import Checkbox from "@material-ui/core/Checkbox";
+import IconButton from "@material-ui/core/IconButton";
+import MoreVertIcon from "@material-ui/icons/MoreVert";
+import Tooltip from "@material-ui/core/Tooltip";
+import i18n from "@dhis2/d2-i18n";
+
+import { isEventCtrlClick, updateSelection } from "./utils/selection";
+import { formatRowValue } from "./utils/formatting";
+import { TableObject, TableColumn } from "./types";
+
+const useStyles = makeStyles({
+    cell: {
+        borderBottom: "2px solid #E0E0E0",
+        minHeight: "55px",
+    },
+});
+
+export interface DataTableBodyProps {
+    rows: TableObject[];
+    columns: TableColumn[];
+    selected: string[];
+    onChange?(newSelection: string[]): void;
+    primaryAction?(rows: TableObject[]): void;
+    openContextualMenu?(row: TableObject, positionLeft: number, positionTop: number): void;
+}
+
+export function DataTableBody(props: DataTableBodyProps) {
+    const {
+        rows,
+        columns,
+        selected,
+        onChange = () => {},
+        primaryAction = () => {},
+        openContextualMenu = () => {},
+    } = props;
+    const classes = useStyles();
+    const isSelected = (name: string) => selected.indexOf(name) !== -1;
+
+    const contextualAction = (row: TableObject, event: React.MouseEvent<unknown>) => {
+        event.stopPropagation();
+        openContextualMenu(row, event.clientY, event.clientX);
+    };
+
+    const handleClick = (event: React.MouseEvent<unknown>, row: any) => {
```

Comment:
Instead of two different castings we should have two methods one for row and another one for checkbox